### DISCUSSION TITLE
adding ssh key as deploy key fails with API, works on Site. (key is invalid. You must supply a key in OpenSSH public key format)

### DISCUSSION THREAD

**Discussion by OlJohnny** [community/community#58320] [Apps, API and Webhooks] [answered]
### Select Topic Area

Question

### Body

Hello all,

i am trying to add my public ssh key via the REST API as a deploy key to one of my repos.
However i get this error:
<code>key is invalid. You must supply a key in OpenSSH public key format</code>
However when i try to add this public key as a deploy key on the github website, it just works.

My public ssh key looks like:
<code>ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIImhgO2rClhx5i01e5bbrOIM+DowzXNZ+NJaqSQXo8fT root@jelly-107</code>

I am currently testing this in Postman.

**Comment by airtower-luna** [comment]
At a glance the format looks correct, except that the comment indicates you're working as root which is a security problem (but shouldn't prevent the key from working).

Exactly how do you try to upload it using the API?

**Comment by OlJohnny** [comment]
the curl command from postman looks like this:

<pre>curl --location --request POST 'https://api.github.com/repos/OlJohnny/repo/keys' \
--header 'Authorization: Bearer github_pat_REDACTED' \
--header 'title: test' \
--header 'key: ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIImhgO2rClhx5i01e5bbrOIM+DowzXNZ+NJaqSQXo8fT root@jelly-107'</pre>

**Comment by airtower-luna** [comment, marked as answer]
The `title` and `key` fields must be part of a JSON payload, not headers. As a `curl` parameter, something like this:

```
-d '{"title":"test","key":"ssh-ed25519 AAA..."}'
```

There's a full example in the documentation: https://docs.github.com/en/rest/deploy-keys?apiVersion=2022-11-28#create-a-deploy-key

Also I suspect it might make sense to remove the comment from the key field (as you're effectively setting a comment with `title`), but I don't know if that's *necessary*.

**Comment by OlJohnny** [comment]
Ahh that makes sense, I think i didn't see the Body Headline. Thanks!
